inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/metrics/GCMetricsRecorder.java

```diff
@@ -0,0 +1,332 @@
+
+    private static final String PAUSE_METRIC_NAME = "pause";
+    private static final String PAUSE_METRIC_FULL_NAME = METRIC_NAME_PREFIX + "pause";
+    private static final String PAUSE_METRIC_DESCRIPTION = "Time spent in GC pause";
+    private static final String PAUSE_UNIT = "ms";
+
+    private static final String MEMORY_PROMOTED_METRIC_NAME = "memory.promoted";
+    private static final String MEMORY_PROMOTED_METRIC_FULL_NAME = METRIC_NAME_PREFIX + "memory/promoted";
+    private static final String MEMORY_PROMOTED_METRIC_DESCRIPTION = "Count of positive increases in the size of the old generation memory pool before GC to after GC";
+    private static final String MEMORY_PROMOTED_UNIT = "bytes";
+
+    private static final String MAX_DATA_SIZE_METRIC_NAME = "max.data.size";
+    private static final String MAX_DATA_SIZE_METRIC_FULL_NAME = METRIC_NAME_PREFIX + "max/data/size";
+    private static final String MAX_DATA_SIZE_METRIC_DESCRIPTION = "Maximum size of old generation memory pool";
+    private static final String MAX_DATA_SIZE_UNIT = "bytes";
+
+    private static final String LIVE_DATA_SIZE_METRIC_NAME = "live.data.size";
+    private static final String LIVE_DATA_SIZE_METRIC_FULL_NAME = METRIC_NAME_PREFIX + "live/data/size";
+    private static final String LIVE_DATA_SIZE_METRIC_DESCRIPTION = "Size of old generation memory pool after a full GC";
+    private static final String LIVE_DATA_SIZE_UNIT = "bytes";
+
+    private static final String MEMORY_ALLOCATED_METRIC_NAME = "memory.allocated";
+    private static final String MEMORY_ALLOCATED_METRIC_FULL_NAME = METRIC_NAME_PREFIX + "memory/allocated";
+    private static final String MEMORY_ALLOCATED_METRIC_DESCRIPTION = "Increase in the size of the young generation memory pool after one GC to before the next";
+    private static final String MEMORY_ALLOCATED_UNIT = "bytes";
+
+    private static final boolean MANAGEMENT_EXTENSIONS_PRESENT = isManagementExtensionsPresent();
+
+    private final NotificationListener notificationListener = this::handleNotification;
+    private GCMetricsSettings config;
+
+    private final TagKey actionTagKey = TagKey.create("action");
+    private final TagKey causeTagKey = TagKey.create("cause");
+
+    private String youngGenPoolName;
+    private String oldGenPoolName;
+    final AtomicLong youngGenSizeAfter = new AtomicLong(0L);
+
+    @Autowired
+    Tagger tagger;
+
+    @Autowired
+    protected SelfMonitoringService selfMonitoringService;
+
+    public GCMetricsRecorder() {
+        super("metrics.gc");
+        if (!MANAGEMENT_EXTENSIONS_PRESENT) {
+            log.info("com.sun.management.GarbageCollectionNotificationInfo is not available on this system, gc metrics are unavailable.");
+        }
+    }
+
+    @Override
+    protected boolean checkEnabledForConfig(MetricsSettings ms) {
+        return ms.getGc().getEnabled().containsValue(true) && MANAGEMENT_EXTENSIONS_PRESENT;
```

Comment:
Small comment: Can we first check MANAGEMENT_EXTENSIONS_PRESENT then we don't have to check